src/system.rs

```diff
@@ -0,0 +1,312 @@
+            pub const PreimageBaseDeposit: Balance = deposit(2, 64);
+            pub const PreimageByteDeposit: Balance = deposit(0, 1);
+            pub const PreimageHoldReason: RuntimeHoldReason = RuntimeHoldReason::Preimage(pallet_preimage::HoldReason::Preimage);
+        }
+
+        impl pallet_preimage::Config for Runtime {
+            type Consideration = frame_support::traits::fungible::HoldConsideration<
+                AccountId,
+                Balances,
+                PreimageHoldReason,
+                frame_support::traits::LinearStoragePrice<
+                    PreimageBaseDeposit,
+                    PreimageByteDeposit,
+                    Balance,
+                >,
+            >;
+            type Currency = Balances;
+            type ManagerOrigin = <$t as SystemConfig>::PreimageOrigin;
+            type RuntimeEvent = RuntimeEvent;
+            /// Rerun benchmarks if you are making changes to runtime configuration.
+            type WeightInfo = weights::pallet_preimage::WeightInfo<Runtime>;
+        }
+
+        parameter_types! {
+            pub const ProxyDepositBase: Balance = deposit(1, 40);
+            pub const AnnouncementDepositBase: Balance = deposit(1, 48);
+            pub const ProxyDepositFactor: Balance = deposit(0, 33);
+            pub const AnnouncementDepositFactor: Balance = deposit(0, 66);
+        }
+
+        /// The type used to represent the kinds of proxying allowed.
+        /// If you are adding new pallets, consider adding new ProxyType variant
+        #[derive(
+            Copy,
+            Clone,
+            Decode,
+            Default,
+            Encode,
+            Eq,
+            MaxEncodedLen,
+            Ord,
+            PartialEq,
+            PartialOrd,
+            RuntimeDebug,
+            TypeInfo,
+        )]
+        pub enum ProxyType {
+            /// Allows to proxy all calls
+            #[default]
+            Any,
+            /// Allows all non-transfer calls
+            NonTransfer,
+            /// Allows to finish the proxy
+            CancelProxy,
+            /// Allows to operate with collators list (invulnerables, candidates, etc.)
+            Collator,
+        }
+
+        impl InstanceFilter<RuntimeCall> for ProxyType {
+            fn filter(&self, c: &RuntimeCall) -> bool {
+                match self {
+                    ProxyType::Any => true,
+                    ProxyType::NonTransfer => !matches!(c, RuntimeCall::Balances { .. }),
+                    ProxyType::CancelProxy => matches!(
+                        c,
+                        RuntimeCall::Proxy(pallet_proxy::Call::reject_announcement { .. })
+                            | RuntimeCall::Multisig { .. }
+                    ),
+                    ProxyType::Collator => {
+                        matches!(c, RuntimeCall::CollatorSelection { .. } | RuntimeCall::Multisig { .. })
+                    }
+                }
+            }
+        }
```

Comment:
There is a problem that introduces coupling between system and consensus macros.
`ProxyType::Collator` assumes that there is a `CollatorSelection`-named pallet implemented on the chain. It is not guaranteed. I understand that `Balances` and `Multisig` are also not guaranteed but IMO `CollatorSelection` is too much of an assumption. 

However, I do not know a better solution.